typing/typecore.ml

```diff
@@ -2605,7 +2719,7 @@ let partial_pred ~lev ~splitting_mode ~allow_modules ?(explode=0)
       } in
   try
     reset_pattern allow_modules;
-    let alloc_mode = simple_pat_mode Value_mode.global in
+    let alloc_mode = simple_pat_mode (Mode.Value.legacy) in
```

Comment:
not sure if this should be `legacy` or `min`

Reply:
I think `min` would be better here.